dist/api_data.py

```diff
@@ -15,10 +15,13 @@ def __init__(self, name, default, desc, subconfig=None, **flags):
     def __cmp__(self, other):
         return cmp(self.name, other.name)
 
-# Metadata shared by all schema objects
-common_meta = [
+common_runtime_config = [
     Config('app_metadata', '', r'''
         application-owned metadata for this object'''),
+]
+
+# Metadata shared by all schema objects
+common_meta = [
```

Comment:
I'd expect to see `common_meta = common_runtime_config + [...` rather than repeating it below.

Is there some case you're trying to exclude where we used to allow `app_metadata` but shouldn't?  Or does that lead to `app_metadata` appearing multiple times somewhere?

Reply:
Yes, `app_metadata` started appearing multiple times at some places. For instance, `app_metadata` is defined over `file_runtime_config`. `file_config` builds upon `file_runtime_config` and `format_meta`, `format_meta` builds upon `common_meta`. If `app_metadata` is part of both `common_meta` and `file_runtime_config`, it shows up twice in `file_config`.

For this reason I had to pull out `app_metadata` from several meta configs and put in back in the final configs as needed.

Reply:
OK, I've taken a closer look and opened WT-4002 to follow up on this.